vault/version_store.go

```diff
@@ -66,6 +69,30 @@ func (c *Core) storeVersionEntry(ctx context.Context, vaultVersion *VaultVersion
 	return true, nil
 }
 
+// storeLastUpgrade will store the last version of Vault that was successfully
+// upgraded and unsealed. This is set on successful startup of Vault after an
+// upgrade. This method is used to set the shutdown behavior for mount entries
+// associated with deprecated builtin plugins.
+func (c *Core) storeLastUpgrade(ctx context.Context, vaultVersion string) error {
+	versionData, err := json.Marshal(vaultVersion)
```

Comment:
If you put this in a struct or map you'd have some flexibility if you ever want to store anything else in this storage entry.